sdk/nodejs/runtime/invoke.ts

```diff
@@ -100,3 +127,135 @@ export async function invoke(tok: string, props: Inputs, opts?: InvokeOptions):
         done();
     }
 }
+
+function invokeSync(tok: string, props: any, opts: InvokeOptions = {}): Promise<any> {
+    const label = `Invoking function: tok=${tok} synchronously`;
+    log.debug(label +
+        excessiveDebugOutput ? `, props=${JSON.stringify(props)}` : ``);
+
+    const obj = gstruct.Struct.fromJavaScript(serializePropertiesSync(props));
+    log.debug(`Invoke RPC prepared: tok=${tok}` + excessiveDebugOutput ? `, obj=${JSON.stringify(obj)}` : ``);
+
+    // Fetch the sync monitor and make an RPC request.
+    const syncInvokes = getSyncInvokes();
+
+    const provider = getProvider(tok, opts);
+
+    let providerRef: string | undefined;
+    if (ProviderRef.isInstance(provider)) {
+        providerRef = provider.getValue();
+    }
+    else if (Resource.isInstance(provider)) {
+        // TODO(cyrusn): issue warning here that we are synchronously blocking an rpc call.
+        providerRef = utils.promiseResult(ProviderRef.get(provider)).getValue();
```

Comment:
Note: @pgavlin @lukehoban we should decide what we want to do here.  Specifically, we can donothing/warn/error/throw, and we can decide what is best on a per env basis.  i.e. we can check for things like catalina/nodejs12 here and whatnot.